enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,498 @@
+
+
+### Drawbacks
+
+> WIP/TBReviewed
+
+- The short-term solution requires engineering effort to implement and stabilize the changes in CCM and other components.
+- The extent of changes in CCM may require significant engineering effort from Red Hat to maintain this functionality. To mitigate it, this feature proposes minimum changes without disrupting existing flow.
+- This approach duplicates some functionality that already exists in the AWS Load Balancer Controller.
+
+## Alternatives (Not Implemented)
+
+> TODO/TBD
+
+### **Defaulting to AWS Load Balancer Controller (ALBC) for the default router**:
+
+While ALBC provides more comprehensive support for NLB security groups, this option was deemed out of scope for the initial goal of minimal changes and addressing immediate customer needs for security issues within the existing CCM framework.
+
+Migrating to ALBC would involve more significant architectural changes and potentially impact existing deployments. However, this remains a viable long-term strategy.
+
+Here is an overall effort:
+- There might need to work out how we manage ALBC: ALBO will be used, or CCM will manage ALBC? Neither ALBO nor ALBC is in payload at this time. Moving either one into payload requires migrating from CPaaS to Prow, requiring approval to add a new component to the core payload; When ALBO was created, the messaging was that all new components should be addons, no new components in the core payload.
+- Migrating from CCM to ALBC is going to require either disruption for customer workloads when delete and recreate the LB, or a huge investment in effort from engineering to re-architect the way is managed router deployments, LBs, and DNS to enable having two ELBs in parallel for the same router deployment, bleeding traffic over, and then deleting the old ELB.
+- Red Hat might need to continue to support using CCM indefinitely for customers who are unwilling to do this migration, so this isn't necessarily just a one-release migration; it would most likely end up supporting all these configurations (CLB with CCM, NLB with CCM, NLB with ALBC) as well as the migration process in perpetuity.
+- Currently there were only scratched the surface of special cases, such as custom security groups, custom DNS, or potential regressions going from CCM to ALBC.
+
+### **Day-2 operations to switch the default router to use ALBC/LBC**:
+
+This would require users to manually deploy and configure ALBC after cluster installation, which does not meet the requirement for an opt-in during initial cluster deployment.
+
+## Open Questions [optional]
+
+> WIP/TODO
+
+- Q: Is the annotation name be changed to include "nlb" in the name? There is no similar annotation in ALBC.
+
+- Q: Is it required to create a KEP to the CCM changes?
```

Comment:
i do not think we need a KEP, but we will need to document this in the aws ccm repo.

Reply:
Thanks for the feedback, documented it. I will also add an action item.